src/newViews/Tagging/Sources/SourcesTable/index.tsx

```diff
@@ -0,0 +1,279 @@
+        const selectColumn: TableColumn<
+            Lead, number, CheckboxProps<number>, CheckboxProps<number>
+        > = {
+            id: 'select',
+            title: '',
+            headerCellRenderer: Checkbox,
+            headerCellRendererParams: {
+                value: selectedIds.length === leadsResponse?.results.length,
+                label: `${selectedIds.length > 0 ? `${selectedIds.length} selected` : 'Select all'}`,
+                onChange: handleSelectAll,
+                indeterminate: !(selectedIds.length === leadsResponse?.results.length
+                || selectedIds.length === 0),
+            },
+            cellRenderer: Checkbox,
+            cellRendererParams: (_, data) => ({
+                name: data.id,
+                value: selectedIds.some(v => v === data.id),
+                onChange: handleSelection,
+            }),
+
+        };
+        const statusColumn: TableColumn<
+            Lead, number, TagProps, TableHeaderCellProps
+        > = {
+            id: 'status',
+            title: 'Status',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRendererClassName: styles.status,
+            cellRenderer: Tag,
+            cellRendererParams: (_, data) => ({
+                actions: statusIconMap[data.status],
+                variant: statusVariantMap[data.status],
+                children: statusLabelMap[data.status],
+            }),
+        };
+        const createdAtColumn: TableColumn<
+            Lead, number, DateOutputProps, TableHeaderCellProps
+        > = {
+            id: 'createdAt',
+            title: 'Created At',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: DateOutput,
+            cellRendererParams: (_, data) => ({
+                value: data.createdAt,
+            }),
+        };
+        const publishedOnColumn: TableColumn<
+            Lead, number, DateOutputProps, TableHeaderCellProps
+        > = {
+            id: 'publishedOn',
+            title: 'Publishing Date',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: DateOutput,
+            cellRendererParams: (_, data) => ({
+                value: data.publishedOn,
+            }),
+        };
+        const publisherColumn: TableColumn<
+            Lead, number, LinkProps, TableHeaderCellProps
+        > = {
+            id: 'sourceDetail',
+            title: 'Publisher',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: Link,
+            cellRendererParams: (_, data) => ({
+                children: data.sourceDetail.title,
```

Comment:
This won't always be there.
```suggestion
                children: data.sourceDetail?.title ?? data.sourceRaw,
```